### PR TITLE
Add priority to database persistence

### PR DESCRIPTION
**Pull request by guperrot**
Please have a look at our [guidelines for contributions](https://github.com/Microsoft/AppCenter-SDK-Android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Add priority field to persistence.

### CODE REVIEW

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/persistence/DatabasePersistenceAndroidTest.java [hunk 1]
```diff
@@ -765,14 +768,26 @@ public void getLogsException() throws PersistenceException, JSONException {
         }
     }
 
+    @NonNull
+    private ContentValues getContentValues(DatabasePersistence persistence, String group) {
```
Comment (jaelim-ms) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/persistence/DatabasePersistenceAndroidTest.java]: Can you move this method to the bottom or top of the file?

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    /**
+     * Mask for persistence within flags.
+     */
+    public static final int PERSISTENCE_MASK = 0xFF;
+
+    /**
+     * An event can be lost due to low bandwidth or disk space constraints.
+     */
+    public static final int PERSISTENCE_NORMAL = 0x01;
```
[the remark this answers is not in the document]
Comment (jaelim-ms) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java]: I'd prefer `0x01`.

**Comment by jaelim-ms** [comment]
We need DEFAULT constant for flag as per spec. If we remove DEFAULT constant from here, it is not a big deal but we should change the spec before making any changes in the code.
https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1193#discussion_r229398988

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java [hunk 3]
```diff
@@ -174,7 +188,10 @@ public boolean onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
                     db.execSQL("ALTER TABLE " + TABLE + " ADD COLUMN `" + COLUMN_TARGET_TOKEN + "` TEXT");
                     db.execSQL("ALTER TABLE " + TABLE + " ADD COLUMN `" + COLUMN_DATA_TYPE + "` TEXT");
                 }
-                db.execSQL("ALTER TABLE " + TABLE + " ADD COLUMN `" + COLUMN_TARGET_KEY + "` TEXT");
+                if (oldVersion < VERSION_TARGET_KEY) {
+                    db.execSQL("ALTER TABLE " + TABLE + " ADD COLUMN `" + COLUMN_TARGET_KEY + "` TEXT");
+                }
+                db.execSQL("ALTER TABLE " + TABLE + " ADD COLUMN `" + COLUMN_PRIORITY + "` INTEGER DEFAULT " + PERSISTENCE_NORMAL);
```
Comment (jaelim-ms) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java]: https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1193#discussion_r229396967

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jwallra) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java]: Personally it's more readable for me too. Admittedly it's explicitly different than how we have it in iOS but if we're following language conventions then I don't mind.

[the remark this answers is not in the document]
Comment (guperrot) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java]: 0x01 is more readable indeed.

**Pull request merged**